zmon_worker_monitor/builtins/plugins/redis_wrapper.py

```diff
@@ -54,19 +57,30 @@ def create(self, factory_ctx):
         if not self.counter_factory:
             self.counter_factory = plugin_manager.get_plugin_obj_by_name('counter', 'Function')
 
-        return propartial(RedisWrapper, counter=self.counter_factory.create(factory_ctx), host=factory_ctx['host'])
+        return propartial(
+            RedisWrapper,
+            counter=self.counter_factory.create(factory_ctx),
+            host=factory_ctx['host'],
+            password=self.__password
+        )
 
 
 class RedisWrapper(object):
     '''Class to allow only readonly access to underlying redis connection'''
 
-    def __init__(self, counter, host, port=6379, db=0, socket_connect_timeout=1, socket_timeout=5):
+    def __init__(self, counter, host, port=6379, db=0, password=None, socket_connect_timeout=1, socket_timeout=5):
         if not host:
             raise ConfigurationError('Redis wrapper improperly configured. Valid redis host is required!')
 
         self._counter = counter('')
         self.__con = redis.StrictRedis(
-            host, port, db, socket_connect_timeout=socket_connect_timeout, socket_timeout=socket_timeout)
+            host,
+            port,
+            db,
+            password,
```

Comment:
It is worth noting that this would be a breaking change for existing checks calling redis with no auth configured and having a password set in the configuration. The solution would be to update the checks to pass `password=None`.
May be it would be helpful to mention that in the docs.

Reply:
Thank you. I went through github search and zmon checks export  results and looks like no one uses password kwarg so it should be safe.

Reply:
The problem will happen once a worker is configured with a redis password, it will be used by all checks not using a password (so all `redis()....` checks will fail)